libs/ngx-dsfr/password/src/lib/password.component.ts

```diff
@@ -116,51 +145,54 @@ export class DsfrPasswordComponent implements ControlValueAccessor, OnInit, OnCh
 
   }
 
-  setAutocompleteAttribute() {
-    const isSignUp = this.type === 'signup';
-    this.autocomplete = isSignUp ? 'new-password' : 'current-password';
+  private setAutocompleteAttribute(): void {
+    this.autocomplete = this.type === 'signup' ? 'new-password' : 'current-password';
   }
 
-  setComponentIds(): void {
-    this.mainLabelId = `${this.id}-input`;
+  private setComponentIds(): void {
+    this.mainInputId = `${this.id}-input`;
     this.ariaDescribedBy = `${this.id}-input-messages`;
     this.showPasswordId = `${this.id}-show`;
+    this.messagesId = `${this.id}-input-messages`;
   }
 
   displayPassword(): void {
     if (this.inputType === 'password') {
       this.inputType = 'text';
+      this.buttonDisplayPasswordText = 'Masquer';
     } else {
       this.inputType = 'password';
+      this.buttonDisplayPasswordText = 'Afficher';
     }
   }
 
-  checkSpecialCharacters() {
+  private checkSpecialCharacters(): void {
 
-    let resultClasse: string = 'info';
+    let resultClass: string = 'info';
     let filteredSpecial: string[] = [];
     const regex = new RegExp(/[^A-Za-z0-9]/);
 
     if (this.value) {
+      console.log(this.value);
       const arr = this.value.split('');
       filteredSpecial = arr.filter((character: string) => regex.test(character));
     }
 
     if (filteredSpecial.length >= this.params.minSpecialCharacters) {
-      resultClasse = 'valid';
+      resultClass = 'valid';
     }
 
-    if (this.hasFailed && !filteredSpecial.length) {
-      resultClasse = 'error';
+    if (this.hasFailed && (!filteredSpecial.length || filteredSpecial.length < this.params.minSpecialCharacters)) {
```

Comment:
Correct me if I'm wrong but `!filteredSpecial.length` is only `true` when it's an empty array ? ( the `length` property will always be present since you initialize `filteredSpecial` with an empty array )
In that case then this condition is already checked by the rest of the OR predicate...
I believe you can simplify it as this : 
`this.hasFailed && filteredSpecial.length < this.params.minSpecialCharacters`